prql-compiler/src/utils/mod.rs

```diff
@@ -85,3 +85,31 @@ impl<T> BreakUp<T> for Vec<T> {
         (self, after)
     }
 }
+
+/// Common operations on iterators over [Option<bool>]
+pub trait IterOfOptBool {
+    /// If any of the items is `None`, this returns `None`.
+    /// If any of the items is `Some(false)`, this returns `Some(false)`.
+    /// Otherwise it returns `Some(true)`.
```

Comment:
To what extent is this just `iter.all(|x| matches!(x, Some(true)))`? I guess the proposed code gives us `Some(true)` rather than the bool.

Reply:
Your example does not do this:
```
/// If any of the items is `None`, this returns `None`.
```